components/purchase_modal/purchase_modal.tsx

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {Stripe, loadStripe} from '@stripe/stripe-js';
+import {Elements} from '@stripe/react-stripe-js';
+
+import {Product} from 'mattermost-redux/types/cloud';
+import {Dictionary} from 'mattermost-redux/types/utilities';
+
+import upgradeImage from 'images/cloud/upgrade.svg';
+import wavesBackground from 'images/cloud/waves.svg';
+import blueDotes from 'images/cloud/blue.svg';
+import LowerBlueDots from 'images/cloud/blue-lower.svg';
+
+import cloudLogo from 'images/cloud/mattermost-cloud.svg';
+
+import RootPortal from 'components/root_portal';
+import FullScreenModal from 'components/widgets/modals/full_screen_modal';
+import {areBillingDetailsValid, BillingDetails} from 'types/cloud/sku';
+import {getNextBillingDate} from 'utils/utils';
+
+import PaymentForm from '../payment_form/payment_form';
+
+import './purchase.scss';
+import 'components/payment_form/payment_form.scss';
+import ProcessPaymentSetup from './process_payment_setup';
+
+const STRIPE_CSS_SRC = 'https://fonts.googleapis.com/css?family=Open+Sans:400,400i,600,600i&display=swap';
+const STRIPE_PUBLIC_KEY = 'pk_test_ttEpW6dCHksKyfAFzh6MvgBj';
+
+const stripePromise = loadStripe(STRIPE_PUBLIC_KEY);
+
+type Props = {
+    show: boolean;
+    isDevMode: boolean;
+    products?: Dictionary<Product>;
+    actions: {
+        closeModal: () => void;
+        getCloudProducts: () => void;
+        completeStripeAddPaymentMethod: (stripe: Stripe, billingDetails: BillingDetails, isDevMode: boolean) => any;
```

Comment:
The return type should be something like `boolean | ServerError`

Reply:
Think we're good here now!